src/smhi/smhi.py

```diff
@@ -78,11 +79,50 @@ def find_stations_from_gps(
 
         user_position = (latitude, longitude)
         self.get_stations(parameter)
-        self.d = []
-
-        all_stations = self.client.stations.data
-        self.d = [
-            s
-            for s in all_stations
-            if distance.distance(user_position, (s["latitude"], s["longitude"])) <= dist
-        ]
+        self.nearby_stations = []
+        all_stations = self.client.stations.stations
+        if dist == 0:
+            stations = [
```

Comment:
I would consider simplifying this or rewriting it as a normal loop. A bit hard to follow complex (multi-line) list comprehensions.